sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -289,19 +408,29 @@ public void dispose() {
         tokenManager.close();
     }
 
-    Mono<Void> send(byte[] bytes, int arrayOffset, int messageFormat) {
+    Mono<Void> sendTransaction(byte[] bytes, int arrayOffset, int messageFormat, AmqpTransaction transactionId) {
+
+        Mono<DeliveryState> workToSend = Mono.create(sink -> sendTransaction(new RetriableWorkItem(bytes,
```

Comment:
In many instances, you are creating a MonoSink. Can all these instances be abstracted? Also, it seems many instances, you follow the `Mono<DeliveryState>` call, with "then". It appears to defeat the purpose of returning a value at all. I'd suggest moving this to `Mono<Void>`.